### PR TITLE
adding peft lora example notebook for ner

### PR DESCRIPTION
**Pull request by JINO-ROHIT**
Hi @BenjaminBossan , 

Ive added a lora example for NER, which is mainly a clone of the huggingface NER notebook by Sylvain here https://huggingface.co/learn/nlp-course/chapter7/2#data-collation with couple changes here and there.

lmk what you think

### CODE REVIEW

**Review by BenjaminBossan — CHANGES_REQUESTED** [review]
Context: #2119

Thanks @JINO-ROHIT for taking this task. Good find of a base notebook to adapt. @HarikrishnanK9 is this what you were looking for?

I haven't done a fine-grained analysis yet, but it looks like the `compute_metrics` are note being used. Therefore, we don't get the interesting metrics defined in there. Could you please check that?

**Comment by JINO-ROHIT** [comment]
will do

**Comment by JINO-ROHIT** [comment]
i added the eval metric part, the recall and precision seem quite low, seems to be the classes are imbalanced, would you like me to handle that?

File: examples/token_classification/peft_lora_ner.ipynb [hunk 1]
```diff
@@ -0,0 +1,1152 @@
...
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "trainable params: 294,912 || all params: 108,021,513 || trainable%: 0.2730\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Configure LoRA (Low-Rank Adaptation) for fine-tuning\n",
+    "peft_config = LoraConfig(target_modules = [\"query\", \"key\"])\n",
```
Comment (BenjaminBossan) [inline examples/token_classification/peft_lora_ner.ipynb:445]:
```suggestion
    "peft_config = LoraConfig(target_modules=[\"query\", \"key\"], task_type=TaskType.TOKEN_CLS))\n",
```

`TaskType` needs to be imported from PEFT.

File: examples/token_classification/peft_lora_ner.ipynb [hunk 2]
```diff
@@ -0,0 +1,1152 @@
...
+      "  warnings.warn(\n"
+     ]
+    }
+   ],
+   "source": [
+    "args = TrainingArguments(\n",
+    "    \"bert-finetuned-ner-lora\",\n",
+    "    evaluation_strategy=\"epoch\",\n",
+    "    save_strategy=\"epoch\",\n",
+    "    learning_rate=2e-5,\n",
```
Comment (BenjaminBossan) [inline examples/token_classification/peft_lora_ner.ipynb:519]: The learning rate should be higher for LoRA. This is probably why you find worse results than expected. I checked  2e-4, 2e-3, and 2e-2, and got best results for 2e-3. However, they were still slightly worse than full fine-tuning and I only ran for 3 epochs, so some further experiments could yield better results still.

File: examples/token_classification/peft_lora_ner.ipynb [hunk 3]
```diff
@@ -0,0 +1,1152 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import required libraries\n",
```
Comment (BenjaminBossan) [inline examples/token_classification/peft_lora_ner.ipynb:9]: Let's add a few sentences to the beginning before starting with the code. Something along the lines of: "In this notebook, we learn how to ...". You can also add the link to the course here.

**Review by BenjaminBossan — CHANGES_REQUESTED** [review]
Thanks a lot for the updates. I checked the notebook and found the source for the low performance, namely the learning rate. Check my comment.

Also, when I first ran this notebook, I got an error:

> ImportError: To be able to use evaluate-metric/seqeval, you need to install the following dependencies['seqeval'] using 'pip install seqeval' for instance'

So let's mention at the beginning which packages the user should install. We can assume that torch and PEFT (and all their dependencies) are installed, but besides `seqeval`, let's also mention `datasets` and `evaluate`.

Moreover, it would be nice to set the batch size to a higher value. The base model is quite small, so even with batch sizes of 32 or more, most users should be able to run the notebook. This will speed up training considerably. You can add a comment that users have to reduce the value if they run OOM.

**Comment by JINO-ROHIT** [comment]
oh yeah, i totally missed the lr part. The metrics look waaay better now, thanks!

**Review by BenjaminBossan — APPROVED** [review]
Thanks for adding this notebook to show NER with PEFT, LGTM.

From my point of view, it's ready to merge. I'll give @HarikrishnanK9 some time to take a look to possibly give more feedback. If we don't hear back by Friday, I'll merge.

**Comment by JINO-ROHIT** [comment]
gotcha

**Comment by HarikrishnanK9** [comment]
@JINO-ROHIT Thank You @BenjaminBossan Just go ahead with the merge.

**Pull request merged**